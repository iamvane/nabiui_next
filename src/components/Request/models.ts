import { number } from "../StudioInstructor/constants"

export interface ApplicationPayload {
  requestId: number;
  userId: string;
  accept: Boolean;
}

export interface StudentType {
  name: string;
  age: number;
}

export interface RequestAvailabilityType {
  day: string;
  timeframe: string;
}

export interface RequestType {
  id?: number;
  createdAt?: string;
  elapsedTime?: string;
  displayName?: string;
  instrument?: string;
  lessonDuration?: string;
  requestTitle?: string;
  requestMessage?: string;
  placeForLessons?: string;
  skillLevel?: string;
  role?: string;
  applicationsReceived?: number;
  applied?: boolean;
  studentDetails?: StudentType[];
  avatar?: string;
  location?: string;
  status?: string;
  availability: RequestAvailabilityType[];
  studentId?: number;
  language?: string;
  timezone?: string;
  gender?: string;
}
