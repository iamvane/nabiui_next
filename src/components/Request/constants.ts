export namespace NewRequestComponent {
  export const pageTitle = 'New Student';
  export const title = 'Trial {instrument} Lesson';
  export const instrumentPlaceholder = '{instrument}';
  export const date = '{dateRepalce} @ {timeReplace}';
  export const datePlaceholder = '{dateRepalce}';
  export const timePlaceholder = '{timeReplace}';
  export const studentDetails = '{studentName}, {studentAge} years old, {skillLevel}';
  export const studentNamePlaceholder = '{studentName}';
  export const agePlaceholder = '{studentAge}';
  export const skillLevelPlaceholder = '{skillLevel}';
  export const lessonDuration = '30 mins';
  export const acceptButton = 'Accept';
  export const passButton = 'Pass';
  export const goToStudioButton = 'Go To Studio';
  export const errorMessage = 'There was an error with your request. Please refresh the page and try again.';
  export const responseMessage = 'Thanks for your response.';

  export const weekdaysLabels = {
    'mon': 'Monday',
    'tue': 'Tuesday',
    'wed': 'Wednesday',
    'thu': 'Thursday',
    'fri': 'Friday',
    'sat': 'Saturday',
    'sun': 'Sunday'
  }
  
  export const timeframeLabels = {
    'early-morning': 'early morning (8am-10am)',
    'late-morning': 'late morning (10am-12pm)',
    'early-afternoon': 'early afternoon (12pm-3pm)',
    'late-afternoon': 'late afternoon (3pm -6pm)',
    'evening': 'evening (6pm-9pm)'
  }

}
