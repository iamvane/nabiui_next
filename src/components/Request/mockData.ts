export const requestsMockData = [
  {
    id: 0,
    applied: false,
    user: {
      id: '0',
      email: 'rox@gmail.com',
      role: 'student',
      firstName: 'Roxana',
      lastName: 'Gallardo',
      displayName: 'Roxana G.',
      distance: '0.5 mi away',
      birthday: '1996-07-12T04:00:00.000Z'
    },
    instrument: 'piano',
    placeForLessons: 'studio',
    lessonDuration: '45',
    location: 'Boston, Ma',
    applications: 17,
    createdAt: '2019-08-05T23:01:25.000Z',
    title: 'Fun and experience piano instructor',
    message: 'Lorem Ipsum is simply dummy text of the printing and typesetting industria',
    students: [
      {
        name: 'Roxana',
        age: 18,
        skillLevel: 'beginner'
      },
    ]
  },
  {
    id: 1,
    applied: true,
    application: {
      createdAt: 'Mon Aug 22, 2019 at 2:00pm EDT',
      rate: 50,
      message: 'Hello Sam, I would love to work with your family and typesetting industria  Ipsum is simply dummy.'
    },
    user: {
      id: '1',
      email: 'sam@gmail.com',
      role: 'parent',
      firstName: 'Sam',
      lastName: 'Flood',
      displayName: 'Sam F.',
      distance: '0.5 mi away'
    },
    instrument: 'piano',
    placeForLessons: 'home',
    lessonDuration: '45',
    createdAt: '2019-08-05T20:19:25.000Z',
    location: 'Boston, Ma',
    applications: 17,
    title: 'Piano teacher needed for my family',
    message: 'Lorem Ipsum is simply dummy text of the printing and typesetting industria',
    students: [
      {
        name: 'Zoe',
        age: 6,
        skillLevel: 'beginner'
      },
      {
        name: 'Esme',
        age: 9,
        skillLevel: 'beginner'
      }
    ]
  },
  {
    id: 2,
    applied: false,
    user: {
      id: '2',
      email: 'victor@gmail.com',
      role: 'parent',
      firstName: 'Victor',
      lastName: 'Charles',
      displayName: 'Victor C.',
      distance: '1 mi away'
    },
    instrument: 'piano',
    placeForLessons: 'home',
    lessonDuration: '45',
    createdAt: '2019-08-05T15:19:25.000Z',
    location: 'Boston, Ma',
    applications: 17,
    title: 'Instructor for one child in Boston',
    message: 'Lorem Ipsum is simply dummy text of the printing and typesetting industria',
    students: [
      {
        name: 'Ethan',
        age: 11,
        skillLevel: 'beginner'
      },
    ]
  }
];
