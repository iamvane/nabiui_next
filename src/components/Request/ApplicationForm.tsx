import * as React from 'react';
import NumberFormat from 'react-number-format';
import {
  Button,
  Grid,
  TextField,
  Typography
} from '@material-ui/core';

import SectionTitle from '../common/SectionTitle';
import { RequestViewComponent } from './constants';

interface Props {
  // TODO: set to RequestType on api integration
  values: any;
  request: any;
  handleChange: (event: React.FormEvent<{}>) => void;
  handleSubmit: (event: React.FormEvent<{}>) => void;
}

function NumberFormatCustom(props: any) {
  const { inputRef, onChange, name,  ...other } = props;

  return (
    <NumberFormat
      {...other}
      ref={inputRef}
      onValueChange={values => {
        onChange({
          currentTarget: {
            value: values.value,
            name: name
          },
        });
      }}
      decimalScale={2}
      fixedDecimalScale={true}
      thousandSeparator={true}
    />
  );
}

const ApplicationForm: React.StatelessComponent<Props> = props => {
  return (
    <React.Fragment>
      <SectionTitle text={RequestViewComponent.sendApplication} />
      <Grid className="nabi-text-center nabi-margin-bottom-small">
        <Typography
          className="nabi-text-mediumbold nabi-margin-top-small nabi-text-uppercase"
        >
          {RequestViewComponent.Labels.LessonRate}
        </Typography>
        <Typography
          className="nabi-margin-bottom-xsmall"
        >
        {RequestViewComponent.lessonDuration.replace(
          RequestViewComponent.lessonDurationPlaceHolder,
          props.request.lessonDuration
        )}
        </Typography>
        <TextField
          className="nabi-rates-field"
          placeholder="$0.00"
          value={(props as any)[props.values.lessonRate]}
          onChange={props.handleChange}
          id={RequestViewComponent.Ids.LessonRate}
          name={RequestViewComponent.FieldNames.LessonRate}
          InputProps={{
            inputComponent: NumberFormatCustom,
          }}
        />
      </Grid>
      <Typography
        className="nabi-text-mediumbold nabi-text-center nabi-margin-bottom-xsmall nabi-text-uppercase"
      >
        {RequestViewComponent.sendMessageTo.replace(
          RequestViewComponent.userPlaceholder,
          props.request.user && props.request.user.firstName
        )}
      </Typography>
      <TextField
        margin="normal"
        className="nabi-margin-remove"
        required={true}
        multiline={true}
        onChange={props.handleChange}
        value={props.values.message}
        id={RequestViewComponent.Ids.Message}
        name={RequestViewComponent.FieldNames.Message}
        fullWidth={true}
        rows={6}
      />
      <Grid item={true} xs={12} className="nabi-text-center">
        <Button
          color="primary"
          onClick={props.handleSubmit}
          variant="contained"
          className="nabi-margin-top-small"
        >
          {RequestViewComponent.sendApplication}
        </Button>
      </Grid>
    </React.Fragment>
  );
};

export default ApplicationForm;
