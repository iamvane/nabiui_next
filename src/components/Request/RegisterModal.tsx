import * as React from 'react';
import Link from 'next/link';
import {
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogContentText,
  DialogTitle
} from '@material-ui/core';

import { Routes } from '../common/constants/Routes';

interface Props {
  isOpen: boolean;
  handleClose: () => void;
}

const RegisterModal: React.StatelessComponent <Props> = props => {
  return (
    <div>
      <Dialog
        open={props.isOpen}
        onClose={props.handleClose}
        aria-labelledby="form-dialog-title"
        scroll="body"
      >
        <DialogTitle id="form-dialog-title nabi-text-uppercase">REGISTER OR LOGIN TO APPLY</DialogTitle>

        <DialogContent>
          <DialogContentText>
            Sign up today and start teaching! Login if you have an account.
          </DialogContentText>
        </DialogContent>

        <DialogActions>
          <Link href={Routes.RegistrationInstructor}>
            <a>
              <Button variant="contained" color="primary">Register</Button>
            </a>
          </Link>
          <Link href={Routes.Login}>
            <a>
              <Button variant="contained" color="secondary" onClick={props.handleClose}>
                Login
              </Button>
            </a>
          </Link>
        </DialogActions>
      </Dialog>
    </div>
  );
};

export default RegisterModal;
