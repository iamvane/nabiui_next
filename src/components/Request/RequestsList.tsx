import * as React from 'react';
import { connect } from 'react-redux';

import { StoreState } from 'redux/store';
import ListTemplate from 'components/common/Templates/ListTemplate';
import { PlaceForLessonsType } from 'components/PlaceForLessons/model';
import RequestFilter from 'components/Request/RequestsFilter';
import Requests from 'components/Request/Requests';
import { RequestsListComponent } from 'components/Request/constants';

interface OwnProps {
}

interface StateProps {
  // TODO: set to RequestType when api integration is done
  requests: any;
}

interface Props extends OwnProps, StateProps {}

interface State extends
  PlaceForLessonsType {
  instrument: string;
  zipCode: string;
  lessonDuration: string;
  sortBy: string;
  skillLevel: string;
}

export class RequestsList extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      instrument: '',
      zipCode: '',
      sortBy: '',
      lessonDuration: 'Any',
      skillLevel: 'Any',
      home: false,
      studio: false,
      online: false,
    };
  }

  public handleChange = (event: React.FormEvent<HTMLInputElement>) => {
    const target = event.currentTarget;
    const value = target.type === 'checkbox' ? target.checked : target.value;
    const name = target.name;

    this.setState({
      ...this.state,
      [name]: value,
    });
  }

  public render() {
    return (
      <div>
        <ListTemplate
          pageTitle={RequestsListComponent.pageTitle}
          results={0}
          handleChange={this.handleChange}
          instrument=""
          address=""
          hasCallToAction={true}
          getLatLng={() => console.log('foo')}
          getLocation={() => console.log('foo')}
          isRequesting={false}
          filterSection={
            <RequestFilter
              instrument={this.state.instrument}
              handleChange={this.handleChange}
              distance={this.state.distance}
              home={this.state.home}
              skillLevel={this.state.skillLevel}
              studio={this.state.studio}
              online={this.state.online}
              zipCode={this.state.zipCode}
              lessonDuration={this.state.lessonDuration}
            />
          }
          mainContent={<Requests requests={this.props.requests} />}
        />
      </div>
    );
  }
}

const mapStateToProps = (state: StoreState, _ownProps: OwnProps): StateProps => {
  return {
    requests: state.requests.requests,
  };
};

export default connect(mapStateToProps, {})(RequestsList);
