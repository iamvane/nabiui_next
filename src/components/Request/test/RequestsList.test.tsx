import * as React from 'react';
import { shallow } from 'enzyme';

import { RequestsList } from '../RequestsList';
import { requestsMockData } from 'components/Request/mockData';

describe('RequestsList', () => {
  const wrapper = shallow(<RequestsList requests={requestsMockData} />);

  it('Matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
