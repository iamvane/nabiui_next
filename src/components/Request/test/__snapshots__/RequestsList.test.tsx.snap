// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequestsList Matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RequestsList
    requests={
      Array [
        Object {
          "applications": 17,
          "applied": false,
          "createdAt": "2019-08-05T23:01:25.000Z",
          "id": 0,
          "instrument": "piano",
          "lessonDuration": "45",
          "location": "Boston, Ma",
          "message": "Lorem Ipsum is simply dummy text of the printing and typesetting industria",
          "placeForLessons": "studio",
          "students": Array [
            Object {
              "age": 18,
              "name": "Roxana",
              "skillLevel": "beginner",
            },
          ],
          "title": "Fun and experience piano instructor",
          "user": Object {
            "birthday": "1996-07-12T04:00:00.000Z",
            "displayName": "Roxana G.",
            "distance": "0.5 mi away",
            "email": "rox@gmail.com",
            "firstName": "Roxana",
            "id": "0",
            "lastName": "Gallardo",
            "role": "student",
          },
        },
        Object {
          "application": Object {
            "createdAt": "Mon Aug 22, 2019 at 2:00pm EDT",
            "message": "Hello Sam, I would love to work with your family and typesetting industria  Ipsum is simply dummy.",
            "rate": 50,
          },
          "applications": 17,
          "applied": true,
          "createdAt": "2019-08-05T20:19:25.000Z",
          "id": 1,
          "instrument": "piano",
          "lessonDuration": "45",
          "location": "Boston, Ma",
          "message": "Lorem Ipsum is simply dummy text of the printing and typesetting industria",
          "placeForLessons": "home",
          "students": Array [
            Object {
              "age": 6,
              "name": "Zoe",
              "skillLevel": "beginner",
            },
            Object {
              "age": 9,
              "name": "Esme",
              "skillLevel": "beginner",
            },
          ],
          "title": "Piano teacher needed for my family",
          "user": Object {
            "displayName": "Sam F.",
            "distance": "0.5 mi away",
            "email": "sam@gmail.com",
            "firstName": "Sam",
            "id": "1",
            "lastName": "Flood",
            "role": "parent",
          },
        },
        Object {
          "applications": 17,
          "applied": false,
          "createdAt": "2019-08-05T15:19:25.000Z",
          "id": 2,
          "instrument": "piano",
          "lessonDuration": "45",
          "location": "Boston, Ma",
          "message": "Lorem Ipsum is simply dummy text of the printing and typesetting industria",
          "placeForLessons": "home",
          "students": Array [
            Object {
              "age": 11,
              "name": "Ethan",
              "skillLevel": "beginner",
            },
          ],
          "title": "Instructor for one child in Boston",
          "user": Object {
            "displayName": "Victor C.",
            "distance": "1 mi away",
            "email": "victor@gmail.com",
            "firstName": "Victor",
            "id": "2",
            "lastName": "Charles",
            "role": "parent",
          },
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Unknown
        address=""
        filterSection={
          <RequestFilter
            distance={undefined}
            handleChange={[Function]}
            home={false}
            instrument=""
            lessonDuration="Any"
            online={false}
            skillLevel="Any"
            studio={false}
            zipCode=""
          />
        }
        getLatLng={[Function]}
        getLocation={[Function]}
        handleChange={[Function]}
        hasCallToAction={true}
        instrument=""
        isRequesting={false}
        mainContent={
          <Requests
            requests={
              Array [
                Object {
                  "applications": 17,
                  "applied": false,
                  "createdAt": "2019-08-05T23:01:25.000Z",
                  "id": 0,
                  "instrument": "piano",
                  "lessonDuration": "45",
                  "location": "Boston, Ma",
                  "message": "Lorem Ipsum is simply dummy text of the printing and typesetting industria",
                  "placeForLessons": "studio",
                  "students": Array [
                    Object {
                      "age": 18,
                      "name": "Roxana",
                      "skillLevel": "beginner",
                    },
                  ],
                  "title": "Fun and experience piano instructor",
                  "user": Object {
                    "birthday": "1996-07-12T04:00:00.000Z",
                    "displayName": "Roxana G.",
                    "distance": "0.5 mi away",
                    "email": "rox@gmail.com",
                    "firstName": "Roxana",
                    "id": "0",
                    "lastName": "Gallardo",
                    "role": "student",
                  },
                },
                Object {
                  "application": Object {
                    "createdAt": "Mon Aug 22, 2019 at 2:00pm EDT",
                    "message": "Hello Sam, I would love to work with your family and typesetting industria  Ipsum is simply dummy.",
                    "rate": 50,
                  },
                  "applications": 17,
                  "applied": true,
                  "createdAt": "2019-08-05T20:19:25.000Z",
                  "id": 1,
                  "instrument": "piano",
                  "lessonDuration": "45",
                  "location": "Boston, Ma",
                  "message": "Lorem Ipsum is simply dummy text of the printing and typesetting industria",
                  "placeForLessons": "home",
                  "students": Array [
                    Object {
                      "age": 6,
                      "name": "Zoe",
                      "skillLevel": "beginner",
                    },
                    Object {
                      "age": 9,
                      "name": "Esme",
                      "skillLevel": "beginner",
                    },
                  ],
                  "title": "Piano teacher needed for my family",
                  "user": Object {
                    "displayName": "Sam F.",
                    "distance": "0.5 mi away",
                    "email": "sam@gmail.com",
                    "firstName": "Sam",
                    "id": "1",
                    "lastName": "Flood",
                    "role": "parent",
                  },
                },
                Object {
                  "applications": 17,
                  "applied": false,
                  "createdAt": "2019-08-05T15:19:25.000Z",
                  "id": 2,
                  "instrument": "piano",
                  "lessonDuration": "45",
                  "location": "Boston, Ma",
                  "message": "Lorem Ipsum is simply dummy text of the printing and typesetting industria",
                  "placeForLessons": "home",
                  "students": Array [
                    Object {
                      "age": 11,
                      "name": "Ethan",
                      "skillLevel": "beginner",
                    },
                  ],
                  "title": "Instructor for one child in Boston",
                  "user": Object {
                    "displayName": "Victor C.",
                    "distance": "1 mi away",
                    "email": "victor@gmail.com",
                    "firstName": "Victor",
                    "id": "2",
                    "lastName": "Charles",
                    "role": "parent",
                  },
                },
              ]
            }
          />
        }
        pageTitle="Requests"
        results={0}
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "address": "",
        "filterSection": <RequestFilter
          distance={undefined}
          handleChange={[Function]}
          home={false}
          instrument=""
          lessonDuration="Any"
          online={false}
          skillLevel="Any"
          studio={false}
          zipCode=""
        />,
        "getLatLng": [Function],
        "getLocation": [Function],
        "handleChange": [Function],
        "hasCallToAction": true,
        "instrument": "",
        "isRequesting": false,
        "mainContent": <Requests
          requests={
            Array [
              Object {
                "applications": 17,
                "applied": false,
                "createdAt": "2019-08-05T23:01:25.000Z",
                "id": 0,
                "instrument": "piano",
                "lessonDuration": "45",
                "location": "Boston, Ma",
                "message": "Lorem Ipsum is simply dummy text of the printing and typesetting industria",
                "placeForLessons": "studio",
                "students": Array [
                  Object {
                    "age": 18,
                    "name": "Roxana",
                    "skillLevel": "beginner",
                  },
                ],
                "title": "Fun and experience piano instructor",
                "user": Object {
                  "birthday": "1996-07-12T04:00:00.000Z",
                  "displayName": "Roxana G.",
                  "distance": "0.5 mi away",
                  "email": "rox@gmail.com",
                  "firstName": "Roxana",
                  "id": "0",
                  "lastName": "Gallardo",
                  "role": "student",
                },
              },
              Object {
                "application": Object {
                  "createdAt": "Mon Aug 22, 2019 at 2:00pm EDT",
                  "message": "Hello Sam, I would love to work with your family and typesetting industria  Ipsum is simply dummy.",
                  "rate": 50,
                },
                "applications": 17,
                "applied": true,
                "createdAt": "2019-08-05T20:19:25.000Z",
                "id": 1,
                "instrument": "piano",
                "lessonDuration": "45",
                "location": "Boston, Ma",
                "message": "Lorem Ipsum is simply dummy text of the printing and typesetting industria",
                "placeForLessons": "home",
                "students": Array [
                  Object {
                    "age": 6,
                    "name": "Zoe",
                    "skillLevel": "beginner",
                  },
                  Object {
                    "age": 9,
                    "name": "Esme",
                    "skillLevel": "beginner",
                  },
                ],
                "title": "Piano teacher needed for my family",
                "user": Object {
                  "displayName": "Sam F.",
                  "distance": "0.5 mi away",
                  "email": "sam@gmail.com",
                  "firstName": "Sam",
                  "id": "1",
                  "lastName": "Flood",
                  "role": "parent",
                },
              },
              Object {
                "applications": 17,
                "applied": false,
                "createdAt": "2019-08-05T15:19:25.000Z",
                "id": 2,
                "instrument": "piano",
                "lessonDuration": "45",
                "location": "Boston, Ma",
                "message": "Lorem Ipsum is simply dummy text of the printing and typesetting industria",
                "placeForLessons": "home",
                "students": Array [
                  Object {
                    "age": 11,
                    "name": "Ethan",
                    "skillLevel": "beginner",
                  },
                ],
                "title": "Instructor for one child in Boston",
                "user": Object {
                  "displayName": "Victor C.",
                  "distance": "1 mi away",
                  "email": "victor@gmail.com",
                  "firstName": "Victor",
                  "id": "2",
                  "lastName": "Charles",
                  "role": "parent",
                },
              },
            ]
          }
        />,
        "pageTitle": "Requests",
        "results": 0,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Unknown
          address=""
          filterSection={
            <RequestFilter
              distance={undefined}
              handleChange={[Function]}
              home={false}
              instrument=""
              lessonDuration="Any"
              online={false}
              skillLevel="Any"
              studio={false}
              zipCode=""
            />
          }
          getLatLng={[Function]}
          getLocation={[Function]}
          handleChange={[Function]}
          hasCallToAction={true}
          instrument=""
          isRequesting={false}
          mainContent={
            <Requests
              requests={
                Array [
                  Object {
                    "applications": 17,
                    "applied": false,
                    "createdAt": "2019-08-05T23:01:25.000Z",
                    "id": 0,
                    "instrument": "piano",
                    "lessonDuration": "45",
                    "location": "Boston, Ma",
                    "message": "Lorem Ipsum is simply dummy text of the printing and typesetting industria",
                    "placeForLessons": "studio",
                    "students": Array [
                      Object {
                        "age": 18,
                        "name": "Roxana",
                        "skillLevel": "beginner",
                      },
                    ],
                    "title": "Fun and experience piano instructor",
                    "user": Object {
                      "birthday": "1996-07-12T04:00:00.000Z",
                      "displayName": "Roxana G.",
                      "distance": "0.5 mi away",
                      "email": "rox@gmail.com",
                      "firstName": "Roxana",
                      "id": "0",
                      "lastName": "Gallardo",
                      "role": "student",
                    },
                  },
                  Object {
                    "application": Object {
                      "createdAt": "Mon Aug 22, 2019 at 2:00pm EDT",
                      "message": "Hello Sam, I would love to work with your family and typesetting industria  Ipsum is simply dummy.",
                      "rate": 50,
                    },
                    "applications": 17,
                    "applied": true,
                    "createdAt": "2019-08-05T20:19:25.000Z",
                    "id": 1,
                    "instrument": "piano",
                    "lessonDuration": "45",
                    "location": "Boston, Ma",
                    "message": "Lorem Ipsum is simply dummy text of the printing and typesetting industria",
                    "placeForLessons": "home",
                    "students": Array [
                      Object {
                        "age": 6,
                        "name": "Zoe",
                        "skillLevel": "beginner",
                      },
                      Object {
                        "age": 9,
                        "name": "Esme",
                        "skillLevel": "beginner",
                      },
                    ],
                    "title": "Piano teacher needed for my family",
                    "user": Object {
                      "displayName": "Sam F.",
                      "distance": "0.5 mi away",
                      "email": "sam@gmail.com",
                      "firstName": "Sam",
                      "id": "1",
                      "lastName": "Flood",
                      "role": "parent",
                    },
                  },
                  Object {
                    "applications": 17,
                    "applied": false,
                    "createdAt": "2019-08-05T15:19:25.000Z",
                    "id": 2,
                    "instrument": "piano",
                    "lessonDuration": "45",
                    "location": "Boston, Ma",
                    "message": "Lorem Ipsum is simply dummy text of the printing and typesetting industria",
                    "placeForLessons": "home",
                    "students": Array [
                      Object {
                        "age": 11,
                        "name": "Ethan",
                        "skillLevel": "beginner",
                      },
                    ],
                    "title": "Instructor for one child in Boston",
                    "user": Object {
                      "displayName": "Victor C.",
                      "distance": "1 mi away",
                      "email": "victor@gmail.com",
                      "firstName": "Victor",
                      "id": "2",
                      "lastName": "Charles",
                      "role": "parent",
                    },
                  },
                ]
              }
            />
          }
          pageTitle="Requests"
          results={0}
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "address": "",
          "filterSection": <RequestFilter
            distance={undefined}
            handleChange={[Function]}
            home={false}
            instrument=""
            lessonDuration="Any"
            online={false}
            skillLevel="Any"
            studio={false}
            zipCode=""
          />,
          "getLatLng": [Function],
          "getLocation": [Function],
          "handleChange": [Function],
          "hasCallToAction": true,
          "instrument": "",
          "isRequesting": false,
          "mainContent": <Requests
            requests={
              Array [
                Object {
                  "applications": 17,
                  "applied": false,
                  "createdAt": "2019-08-05T23:01:25.000Z",
                  "id": 0,
                  "instrument": "piano",
                  "lessonDuration": "45",
                  "location": "Boston, Ma",
                  "message": "Lorem Ipsum is simply dummy text of the printing and typesetting industria",
                  "placeForLessons": "studio",
                  "students": Array [
                    Object {
                      "age": 18,
                      "name": "Roxana",
                      "skillLevel": "beginner",
                    },
                  ],
                  "title": "Fun and experience piano instructor",
                  "user": Object {
                    "birthday": "1996-07-12T04:00:00.000Z",
                    "displayName": "Roxana G.",
                    "distance": "0.5 mi away",
                    "email": "rox@gmail.com",
                    "firstName": "Roxana",
                    "id": "0",
                    "lastName": "Gallardo",
                    "role": "student",
                  },
                },
                Object {
                  "application": Object {
                    "createdAt": "Mon Aug 22, 2019 at 2:00pm EDT",
                    "message": "Hello Sam, I would love to work with your family and typesetting industria  Ipsum is simply dummy.",
                    "rate": 50,
                  },
                  "applications": 17,
                  "applied": true,
                  "createdAt": "2019-08-05T20:19:25.000Z",
                  "id": 1,
                  "instrument": "piano",
                  "lessonDuration": "45",
                  "location": "Boston, Ma",
                  "message": "Lorem Ipsum is simply dummy text of the printing and typesetting industria",
                  "placeForLessons": "home",
                  "students": Array [
                    Object {
                      "age": 6,
                      "name": "Zoe",
                      "skillLevel": "beginner",
                    },
                    Object {
                      "age": 9,
                      "name": "Esme",
                      "skillLevel": "beginner",
                    },
                  ],
                  "title": "Piano teacher needed for my family",
                  "user": Object {
                    "displayName": "Sam F.",
                    "distance": "0.5 mi away",
                    "email": "sam@gmail.com",
                    "firstName": "Sam",
                    "id": "1",
                    "lastName": "Flood",
                    "role": "parent",
                  },
                },
                Object {
                  "applications": 17,
                  "applied": false,
                  "createdAt": "2019-08-05T15:19:25.000Z",
                  "id": 2,
                  "instrument": "piano",
                  "lessonDuration": "45",
                  "location": "Boston, Ma",
                  "message": "Lorem Ipsum is simply dummy text of the printing and typesetting industria",
                  "placeForLessons": "home",
                  "students": Array [
                    Object {
                      "age": 11,
                      "name": "Ethan",
                      "skillLevel": "beginner",
                    },
                  ],
                  "title": "Instructor for one child in Boston",
                  "user": Object {
                    "displayName": "Victor C.",
                    "distance": "1 mi away",
                    "email": "victor@gmail.com",
                    "firstName": "Victor",
                    "id": "2",
                    "lastName": "Charles",
                    "role": "parent",
                  },
                },
              ]
            }
          />,
          "pageTitle": "Requests",
          "results": 0,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
